core/src/hir/attrs.rs

```diff
@@ -750,6 +750,8 @@ pub struct BackendAttrSupport {
     pub utf8_strings: bool,
     /// Whether the language uses UTF-16 strings
     pub utf16_strings: bool,
+    /// Whether the language supports returning or using slices with 'static lifetimes.
+    pub static_slices: bool,
```

Comment:
nit: HIR validation should throw an error if this isn't set and it encounters a static slice

Reply:
Done, I've had to add a check to see if a method is disabled to support this for params, though.

Didn't do this for field types because there's already a check to see if structs are disabled. Not sure if we want to support disabling fields based on the backend though.